models/token.js

```diff
@@ -0,0 +1,115 @@
+'use strict';
+
+const Joi = require('joi');
+const mutate = require('../lib/mutate');
+
+const MODEL = {
+    token: Joi
+        .string()
+        .token(),
+
+    id: Joi
+        .number()
+        .integer()
+        .positive(),
+
+    user: Joi
+        .number()
+        .integer()
+        .positive()
+        .description('User ID'),
+
+    name: Joi
+        .string()
+        .max(128)
+        .description('Token name')
+        .example('Mobile token'),
+
+    description: Joi
+        .string()
+        .max(256)
+        .description('Token description')
+        .example('Used to authenticate the mobile app'),
+
+    lastUsed: Joi
+        .date()
+        .allow(null)
+        .description('Last used')
+};
+
+module.exports = {
+    /**
+     * All the available properties of Token
+     *
+     * @property base
+     * @type {Joi}
+     */
+    base: Joi.object(MODEL).label('Token'),
+
+    /**
+     * Properties for token that will come back during a GET request
+     *
+     * @property get
+     * @type {Joi}
+     */
+    get: Joi
+        .array()
+        .items(Joi.object(mutate(MODEL, [
+            'id',
+            'name',
+            'lastUsed'
+        ], [
+            'description'
+        ]))).label('Get tokens'),
+
+    /**
+     * Properties for Token that will be passed during a CREATE request
+     *
+     * @property create
+     * @type {Joi}
+     */
+    create: Joi.object(mutate(MODEL, [
+        'user',
+        'name'
+    ], [
+        'description'
+    ])).label('Create token'),
+
+    /**
+     * Properties for token that will be passed during a UPDATE requeste
+     *
+     * @property update
+     * @type {Joi}
+     */
+    update: Joi.object(mutate(MODEL, [
+        'id'
+    ], [
+        'name',
+        'description',
+        'lastUsed'
```

Comment:
The `lastUsed` attribute is intended to describe when the token was last accessed. Why do we want the caller to be able to update it?